tests/key.go

```diff
@@ -134,9 +137,29 @@ func (tp *TestSuite) TestGenerate(t *testing.T) {
 		t.Errorf("expected the key to be called 'foo', got '%s'", k.Name())
 	}
 
-	if !strings.HasPrefix(k.Path().String(), "/ipns/Qm") {
-		t.Errorf("expected the key to be prefixed with '/ipns/Qm', got '%s'", k.Path().String())
+	verifyIPNSPath(t, k.Path().String())
+}
+
+func verifyIPNSPath(t, t *testing.T, p string) bool {
```

Comment:
This way of doing validation won't tell us where the bad string actually came from, instead we'll get errors in the middle of this function and have to figure out what caused it.

Two ways to solve this:
1) Add `t.Helper()` as the first line in this function, which will make test errors that are output be tied to the line number of the caller
2) Throw the error in the caller instead of in the method